share/picongpu/unit/dimensional_tests/MovingWindow.cpp

```diff
@@ -0,0 +1,209 @@
+/* Copyright 2025 Tapish Narwal
+ *
+ * This file is part of PIConGPU.
+ *
+ * PIConGPU is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * PIConGPU is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with PIConGPU.
+ * If not, see <http://www.gnu.org/licenses/>.
+ */
+
+#include <pmacc/boost_workaround.hpp>
+
+#include <pmacc/test/PMaccFixture.hpp>
+
+#include <cmath>
+
+#include <catch2/catch_approx.hpp>
+#include <catch2/catch_test_macros.hpp>
+#include <catch2/generators/catch_generators.hpp>
+#include <picongpu/defines.hpp>
+#include <picongpu/simulation/control/MovingWindow.hpp>
+
+using namespace picongpu;
+
+//! Helper to setup the PMacc environment
+using TestFixture = pmacc::test::PMaccFixture<simDim>;
+static TestFixture pmaccFixture;
+
+struct MovingWindowTestFixture
+{
+    MovingWindow& window;
+
+    MovingWindowTestFixture() : window(MovingWindow::getInstance())
+    {
+        window.resetForTesting();
+    }
+
+    ~MovingWindowTestFixture()
+    {
+        window.resetForTesting();
+    }
+};
+
+TEST_CASE("unit::MovingWindow_origin", "[movingWindow test]")
+{
+    MovingWindowTestFixture fixture;
+    auto& movingWindow = fixture.window;
+
+    pmaccFixture.initGrids(
+        pmacc::DataSpace<3>{500, 1000, 500}.shrink<simDim>(),
+        pmacc::DataSpace<3>{100, 100, 100}.shrink<simDim>(),
+        pmacc::DataSpace<3>{0, 0, 0}.shrink<simDim>());
+
+    auto const globalWindowSizeInMoveDirection = 900;
+
+    auto const c = static_cast<float_64>(sim.pic.getSpeedOfLight());
+    auto const dt = static_cast<float_64>(sim.pic.getDt());
+    // should both be 1 for correctness of the tests
+    assert(c == 1. && dt == 1.);
+
+    auto const cellSize = sim.pic.getCellSize();
+
+    constexpr auto moveDirection = MovingWindow::moveDirection;
+    auto const cellSizeInMoveDirection = static_cast<float_64>(cellSize[moveDirection]);
+    auto const cellsPerStep = 1. / cellSizeInMoveDirection;
```

Comment:
I would probably factor out this whole initialisation into an own function. The reader quickly skimming this to find out about a particular behaviour doesn't need to bother how precisely `cellsPerStep` was computed. In fact, they don't even need to see `c` and `dt` at all.
```suggestion
    auto [globalWindowSizeInMoveDirection, ...] = init();
```

Reply:
I see your point but am not particularly enthusiastic about this, since I think if someone wants to start reasoning about the logic of the tests they might care if these values are calculated correctly. Also a long structured binding isn't exactly pretty either. I do agree `c` and `dt` can be completely hidden, but I am for keeping it as it is to not add indirection. 
So unless you really want me to change this, I am going to leave it